### PR TITLE
Trail mark

### PR DESCRIPTION
**Pull request by bianchi-dy**
Referencing #23, changes in this PR:

**Trail properties**

* Trail mark implemented as one area where the path is dicated by adding/subtracting the stroke width for the corresponding coordinate (see figure below) – the path is generated using `d3.line().curve(curveCardinal.tension(0.96))` to get a path similar to Multiline
* Uses `stroke-width` to dictate variable 'stroke'
* Uses `fillOpacity` and `fill` to dictate opacity and color
* `sort` and `close` are enabled

**Trail docs**

* Updated trail docs

**Feedback needed**

* I use a lot of custom logic to implement Trail due to sorting the points and aesthetics together, plus generating the coordinates for the path. Perhaps there are more parts that can be adapted to reuse present code in Path.js (the primary mixin), that are not obvious to me at the moment
* I'm not sure how 'smoothly' the trail looks like on other screens, as the end points of the line are sometimes not rounded due to how the entire path is generated – right now my current implementation is as follows:

1. Convert points to screen coordinates 
2. Get equation of line from current point to next point -> eqn 1 with unit vector 1
3. Find unit vector parallel to unit vector 1 -> unit vector 2 
4. Find points +/- stroke width/2 along unit vector 2 from current point
5. Organize array of path coordinates such that top section is followed by bottom section of path
6. Draw path using curveCardinal

And it ends up looking like this:

screenshot 2019-02-25 at 6 24 26 pm

### CODE REVIEW

File: docs/.vuepress/components/SymbolMarkDemo.vue [hunk 1]
```diff
@@ -102,7 +98,7 @@ export default {
   computed : {
     fill () {
       if (this.color === 'both' || this.color === 'fill') {
-        return { scale: { scale: 'viridis', variable: 'explanatory' } }
+        return { get: 'explanatory', scale: { type: 'viridis', domain: 'explanatory' } }
```
Comment (luucvanderzee) [inline docs/.vuepress/components/SymbolMarkDemo.vue]: This syntax is not used anymore... we now use `val` which gets the value directly through the scoped slot

File: docs/.vuepress/components/ScatterplotMatrix.vue [hunk 2]
```diff
@@ -0,0 +1,370 @@
+<template>
```
Comment (luucvanderzee) [inline docs/.vuepress/components/ScatterplotMatrix.vue]: Wow this is a big one lol. This can be improved a lot with the new `vgg-repeat` component, especially after I finish the automatic axis positioning (is currently under construction on my `section-slots` branch).

File: docs/.vuepress/components/Cartesian.vue [hunk 3]
```diff
@@ -0,0 +1,145 @@
+<template>
```
Comment (luucvanderzee) [inline docs/.vuepress/components/Cartesian.vue]: What is this?

File: docs/.vuepress/components/ScatterplotMatrix.vue (hunk 2 above)
Comment (luucvanderzee) [reply docs/.vuepress/components/ScatterplotMatrix.vue]: Also, when I run `yarn docs:dev`, I get an compilation error due to this component not having one root element...

File: package.json [hunk 4]
```diff
@@ -23,6 +23,7 @@
     "d3-scale-chromatic": "^1.3.3",
     "d3-shape": "^1.2.2",
     "d3-time-format": "^2.1.3",
+    "docs": "^0.0.0-37",
```
Comment (luucvanderzee) [inline package.json]: ?

**Comment by luucvanderzee** [comment]
There is a lot stuff from the Axis options branch in here, which we didn't merge into master yet because it is not ready... and a lot of random things like the Scatterplot matrix component, and some old obsolete code seems to have returned (like `mapAesthetics.js`). Is it possible to clean this branch up a little, so that we can merge in just the Trail mark? If it works more or less like the other mark, it should be quite self-contained, so isolating the Trail functionality shouldnt be too hard

File: docs/.vuepress/components/ScatterplotMatrix.vue (hunk 2 above)
Comment (bianchi-dy) [reply docs/.vuepress/components/ScatterplotMatrix.vue]: I think this is leftover from when I was trying to figure out how to merge branches and merged the wrong ones. Thanks for spotting this

File: docs/.vuepress/components/Cartesian.vue (hunk 3 above)
Comment (bianchi-dy) [reply docs/.vuepress/components/Cartesian.vue]: Leftover from my previous faulty merge attempts, deleting this

**Comment by bianchi-dy** [comment]
Okay, I think the commits from today cover what you've mentioned @luucvanderzee 

**Changes**

- Removed leftovers from faulty merge (i.e. obsolete code like mapAesthetics.js, other branch stuff like BaseAxis.js and so on) + random bits unrelated to trail mark
- Added demo to trail mark documentation

File: src/components/Marks/Label.vue [hunk 5]
```diff
@@ -1,80 +1,65 @@
 <script>
```
Comment (luucvanderzee) [inline src/components/Marks/Label.vue]: Why did you remove all whitespace from this file?

File: src/mixins/Marks/Path.js [hunk 6]
```diff
@@ -48,7 +48,7 @@ export default {
     },
 
     strokeWidth: {
-      type: Number,
+      type: [Number, Array, Object, undefined],
```
Comment (luucvanderzee) [inline src/mixins/Marks/Path.js]: I am assuming the `Array` is for an array of `strokeWidth` values for the trail mark, is that correct? But why do we need the `Object`? Also, it is probably better to just keep this `type: Number` and overwrite this in the `Trail` mark, so that only the trail mark accepts an array of values.

File: src/components/Marks/Label.vue (hunk 5 above)
Comment (bianchi-dy) [reply src/components/Marks/Label.vue]: I did a bunch of reverts for the unrelated-to-trail-mark code, this was unintentional
